src/transforms/math.ts

```diff
@@ -27,13 +28,25 @@ function replaceEqnarray(log: Logger, value: string, file: string) {
     .replace(/\\end{eqnarray}/g, '\\end{align*}');
 }
 
+type RenderResult = { html?: string; warnings?: string[]; error?: string };
+
+function removeWarnings(result: RenderResult, predicate: (warning: string) => boolean) {
+  if (!result.warnings) return result;
+  const nextWarnings = result.warnings.filter(predicate);
+  if (nextWarnings.length === 0) return { html: result.html };
```

Comment:
This would lose `result.error` if it exists. Looks like we maybe never get that case, i.e. `RenderResult` is either `html/warnings` only or `error` only? But from this function that's not clear.

Reply:
Nice, yeah, yours is better!